repos/system_upgrade/common/actors/livemode/modify_userspace_for_livemode/libraries/prepareliveimage.py

```diff
@@ -0,0 +1,515 @@
+        api.current_logger().error(msg)
+        details = {'Problem': 'Failed to open target userspace /etc/passwd for reading'}
+        raise StopActorExecutionError(
+            'Could not set up passwordless root login for the upgrade environment.',
+            details=details
+        )
+
+    new_passwd_lines = []
+    found_root_entry = False
+    for entry in passwd:
+        if entry.startswith('root:'):
+            found_root_entry = True
+
+            root_fields = entry.split(':')
+            root_fields[1] = ''
+            entry = ':'.join(root_fields)
+
+        new_passwd_lines.append(entry)
+
+    if not found_root_entry:
+        msg = 'Failed to set up a passwordless root login in the target userspace - there is no root user entry.'
+        api.current_logger().warning(msg)
+        details = {'Problem': 'There is no root user entry in target userspace\'s /etc/passwd'}
+        raise StopActorExecutionError(
+            'Could not set up passwordless root login for the upgrade environment.',
+            details=details
+        )
+
+    try:
+        with context.open('/etc/passwd', 'w+') as passwd_file:
+            passwd_contents = ''.join(new_passwd_lines)
+            passwd_file.write(passwd_contents)
+    except OSError:
+        api.current_logger().warning('Failed to write new contents into target userspace /etc/passwd.')
+        raise StopActorExecutionError(
+            'Could not set up passwordless root login for the upgrade environment.',
+            details={'Problem': 'Filed to write /etc/passwd for the upgrade environment'}
+        )
+
+
+def enable_dbus(context):
+    """
+    Enable dbus-daemon into the target userspace
+    Looks like it's not enabled by default when installing into a container.
+    """
+    api.current_logger().info('Configuring the dbus services')
+
+    links = ['/etc/systemd/system/multi-user.target.wants/dbus-daemon.service',
+             '/etc/systemd/system/dbus.service',
+             '/etc/systemd/system/messagebus.service']
+
+    for link in links:
+        try:
+            os.symlink('/usr/lib/systemd/system/dbus-daemon.service', context.full_path(link))
+        except OSError as err:
+            details = {'Problem': 'An error occurred while creating the systemd symlink', 'source_error': str(err)}
+            raise StopActorExecutionError('Cannot enable the dbus services', details=details)
+
+
+def setup_network(context, enable_nm):
+    """
+    Setup network for the livemode image.
+
+    Copy ifcfg files and NetworkManager's system connections into the live image.
+    :returns: None
+    :raises StopActorExecutionError: The exception is raised when failing to copy the configuration
+                                     files into the livemode image.
+    """
+    # TODO(mhecko): implementation here is incomplete
+    # ideally we'd need to run 'nmcli con migrate' for the live mode.
+    if not enable_nm or get_target_major_version() < "9":
```

Comment:
This is a bug.  The caller is only sending `context`, not `enable_nm`.  I think this is the correct fix, but please double check since I may not have understood it correctly:

```suggestion
def setup_network(context):
    """
    Setup network for the livemode image.

    Copy ifcfg files and NetworkManager's system connections into the live image.
    :returns: None
    :raises StopActorExecutionError: The exception is raised when failing to copy the configuration
                                     files into the livemode image.
    """
    # TODO(mhecko): implementation here is incomplete
    # ideally we'd need to run 'nmcli con migrate' for the live mode.
    if get_target_major_version() < "9":
```